src/Microsoft.IdentityModel.Tokens/TokenValidationResult.cs

```diff
@@ -36,7 +60,36 @@ public class TokenValidationResult
         /// <summary>
         /// The <see cref="ClaimsIdentity"/> created from the validated security token.
         /// </summary>
-        public ClaimsIdentity ClaimsIdentity { get; set; }
+        public ClaimsIdentity ClaimsIdentity
+        {
+            get
+            {
+                if (_claimsIdentity == null)
+                    _claimsIdentity = CreateClaimsIdentity();
+
+                return _claimsIdentity;
+            }
+            set
```

Comment:
>set [](http://example.com/codeflow?start=12&length=3)

So a ClaimsIdentity can be created via CreateClaimsIdentity() and be set directly?

Reply:
This is for back-compat